File: P_1496_火烧赤壁.py
n = int(input())
a, b = [], []
for _ in range(n):
    A, B = map(int, input().split())
    a.append(A)
    b.append(B)
a.sort()
b.sort()
ans = 0
for i in range(n):
    ans += b[i] - a[i]
    if i and a[i] < b[i - 1]:
        ans -= b[i - 1] - a[i]
print(ans)

File: P_1496_火烧赤壁.cpp
#include <bits/stdc++.h>
using namespace std;

#define ing long long

signed main() {
    int n, ans = 0;
    cin >> n;
    vector<int> a, b;
    for (int i = 0; i < n; ++i) {
        int A, B;
        cin >> A >> B;
        a.emplace_back(A);
        b.emplace_back(B);
    }
    sort(a.begin(), a.end());
    sort(b.begin(), b.end());
    for (int i = 0; i < n; ++i) {
        ans += b[i] - a[i];
        if (i and a[i] < b[i - 1]) {
            ans -= b[i - 1] - a[i];
        }
    }
    cout << ans;
}
